extensions/amp-video-docking/0.1/viewport-rect.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {LayoutRectDef, layoutRectLtwh} from '../../../src/layout-rect';
+
+/**
+ * @param {function():T} get
+ * @return {{get: function():T, configurable: boolean, enumerable: boolean}}
+ * @template T
+ */
+const readonlyGetterProp = get => ({
```

Comment:
Hmm can you elaborate on why this is needed?

Reply:
Viewport box is not static, so in order to get a `Rect`-like box using the viewport, you need readonly getters :)

Reply:
If the question is why I added this general concept, the answer is internal simplicity. It's a single object and it's quite cheap.